packages/modules/common/modbus.py

```diff
@@ -79,23 +79,24 @@ def divide_rounding_up(numerator: int, denominator: int):
             response = read_register_method(
                 address, number_of_addresses, **kwargs)
             if response.isError():
-                raise response
+                raise FaultState(__name__+" "+str(type(response))+" " +
+                                 str(response), FaultStateLevel.ERROR)
             decoder = BinaryPayloadDecoder.fromRegisters(
                 response.registers, Endian.Big if big_endian else Endian.Little)
             result = [getattr(decoder, t.decoding_method)() for t in types]
             return result if multi_request else result[0]
         except pymodbus.exceptions.ConnectionException as e:
-            raise ModuleError("TCP-Client konnte keine Verbindung zu " + str(self.address) + ":" + str(self.port) +
-                              "aufbauen. Bitte Einstellungen (IP-Adresse, ..) und " + "Hardware-Anschluss pruefen.",
-                              ModuleErrorLevel.ERROR) from e
+            raise FaultState("TCP-Client konnte keine Verbindung zu " + str(self.address) + ":" + str(self.port) +
+                             " aufbauen. Bitte Einstellungen (IP-Adresse, ..) und " + "Hardware-Anschluss pruefen.",
+                             FaultStateLevel.ERROR) from e
```

Comment:
Wie schon im vorhergehenden größeren Kommentar geschrieben emphele ich wärmstens, sowas in spezielle Klassen auszulagern. Dann wird dieser Aufruf hier zu:
```python
raise ModbusTcpClientConnectionError(self.address, self.port) from e
```
Und damit übersichtlicher. Den String kann die Klasse `ModbusTcpClientConnectionError` selbst erzeugen. Im simpelsten Fall sieht die Klasse so aus:
```python
class ModbusTcpClientConnectionError(FaultState):
    def __init__(self, host: str, port: int):
        super().__init__("Modbus-TCP-Client keine Verbindung zu " + host + " und so weiter", FaultStateLevel.ERROR)
```
Im komplexeren Fall behält die Klasse auch noch weitere Eigenschaften, die serialisiert werden, wie in meinem vorherigen Kommentar zu strukturieren Exceptions angeregt.